libcaf_net/caf/net/stream_transport.hpp

```diff
@@ -61,18 +66,35 @@ class stream_transport {
     return handle_;
   }
 
+  actor_system& system() {
+    return manager().system();
+  }
+
+  application_type& application() {
+    return worker_.application();
+  }
+
+  transport_type& transport() {
+    return *this;
+  }
+
+  endpoint_manager& manager() {
+    return *manager_;
+  }
+
   // -- member functions -------------------------------------------------------
 
   template <class Parent>
```

Comment:
Is the `Parent` template and parameter still needed even though you the getters for `endpoint_managers` and so on above?

Reply:
Especially if you are using the specific type `endpoint_manager*` on line 237. This template seems especially odd..

Reply:
Yes, because `Parent` is the actual type, i.e., `endpoint_manager_impl<...>` and we might want to wrap the transport one day.

Reply:
Oh yes, I see that! 👍